pixeltable/catalog/catalog.py

```diff
@@ -59,61 +122,177 @@ def __init__(self) -> None:
         self._tbls = {}  # don't use a defaultdict here, it doesn't cooperate with the debugger
         self._init_store()
 
-    def get_dir_path(self, dir_id: UUID) -> str:
+    def get_dir_path(self, dir_id: UUID) -> Path:
         """Return path for directory with given id"""
-        session = env.Env.get().session
+        dir_id_ = dir_id
+        conn = Env.get().conn
         names: list[str] = []
         while True:
-            dir = session.query(schema.Dir).filter(schema.Dir.id == dir_id).one()
+            q = sql.select(schema.Dir).where(schema.Dir.id == dir_id_)
+            # _debug_print(for_update=False, msg=f'dir id={dir_id}')
+            row = conn.execute(q).one()
+            dir = schema.Dir(**row._mapping)
             if dir.md['name'] == '':
                 break
             names.insert(0, dir.md['name'])
-            dir_id = dir.parent_id
+            dir_id_ = dir.parent_id
```

Comment:
What's the relationship between `dir_id` and `dir_id_` here? Seems strange do be doing this and then asserting `dir_id` is not none?

Reply:
Fixed.